peer-to-peer-scheduler/peer-to-peer-scheduler.gs

```diff
@@ -0,0 +1,827 @@
+function cancelEvents() {
+  let setupResult = initSetup();
+  if (!setupResult.success) {
+    return;
+  }
+  let { calendar, activeSheet, oneToOneData } = setupResult;
+  let calendarId = calendar.getId();
+
+  let canceledRows = [];
+  let invalidMeetings = [];
+  let partialCanceledRows = [];
+  for (let rowIndex = 0; rowIndex < oneToOneData.length; ++rowIndex) {
+    let rowNumber = rowIndex + DATA_BEGIN_ROW;
+    let opsRequestFlag = parseInt(activeSheet.getRange(rowNumber, OPS_REQUEST_INDEX + 1).getValue());
+    if (opsRequestFlag !== FLAG_CANCEL_INVITE) {
+      // This row doesn't need to be canceled
+      continue;
+    }
+
+    let meetIdCell = activeSheet.getRange(rowNumber, MEETING_ID_INDEX + 1);
+    let meetLinkCell = activeSheet.getRange(rowNumber, MEETING_LINK_INDEX + 1);
+    let meetId = meetIdCell.getValue().trim();
+    if (meetId === "") {
+      // Meeting is not set
+      continue;
+    }
+
+    let event = getEvent(calendar, meetId);
+    if (!event) {
+      continue;
+    }
+    
+    let cancelDatesCell = activeSheet.getRange(rowNumber, CANCELLATION_DATES_INDEX + 1);
+    let cancelDates = oneToOneData[rowIndex][CANCELLATION_DATES_INDEX].trim();
+
+    if (cancelDates === "") { // Full Invite cancellation
+      try {
+        // If the event was canceled previously (e.g. someone manually canceled), then deleteEvent() throws exception.
+        event.deleteEvent();
+      } catch (error) {
+        logBeinDelimiter();
+        console.log(ERROR_CODE_COULD_NOT_CANCEL_EVENT);
+        console.log(error);
+        logEndDelimiter();
+        invalidMeetings.push(rowNumber);
+        continue;
+      }
+      canceledRows.push(rowNumber);
+      meetIdCell.clearContent();
+      meetLinkCell.clearContent();
+      continue;
+    }
+ 
+    let cancelDatesArray = cancelDates.split(",");
+    for (let i = 0; i < cancelDatesArray.length; ++i) {
+      if(!parseDate(cancelDatesArray[i].trim())) {
+        continue;
+      }
+    }
+    let cancelSuccessDates = [];
+    try {
+      // If the event was canceled previously (e.g. someone manually canceled), then deleteEvent() throws exception.
+      let instancesDates = {} // Each instance id format is "meetId_yyyymmddThhmmssZ"
+      Calendar.Events.instances(calendarId, meetId).items.forEach(
+        function (instance) {
+          currentDate = instance.id.split("_")[1].substring(0,8)
+          instancesDates[currentDate] = instance.id;
+        }
+      );
+      for (let i = 0; i < cancelDatesArray.length; ++i) {
+        let currentDate = cancelDatesArray[i].trim().split(DATE_DELIMITER).join("");
+        if (currentDate in instancesDates) {
+          let event = getEvent(calendar, instancesDates[currentDate]);
+          if (!event) {
+            continue;
+          }
+          event.deleteEvent();
+          cancelSuccessDates.push(cancelDatesArray[i]);
+        }
+      }
+      let instancesCount = Calendar.Events.instances(calendarId, meetId).items.length;
```

Comment:
`Calendar.Events.instances` is an `Calendar API`. `Calendar API` calls take relatively more time. So do not use them indiscriminately. If you want to know all the events are canceled, you can compute it while cancelling the events - by counting how many events are canceled.

Reply:
- removed the API call and counted using `cancelSuccessDates` array's length and comparing it with `instancesArray`'s length